### DISCUSSION TITLE
Touch screen support for M5Stack Core2 devices?

### DISCUSSION THREAD

**Discussion by chhorisberger** [chhorisberger/M5StackMenuSystem#6] [General]
For context, I initally wrote this library for (first generation) M5Stack devices which only featured three physical buttons in the front. 

But recently people have been asking me if support for the (second generation) M5Stack Core2 devices would be possible.

Thanks to @lvlhead, who ported the library to M5Unified, it can now run on M5Stack Core2 devices as well.

Since then, I obviously couldn't resist ordering a M5Stack Core2 devices for myself as well ;)

But when playing around with it, I noticed that the built-in (non-physical) capacitive keys used there make the user experience somewhat more irritating compared to the first generation M5Stack devices.

Since I already had a device now, I did first push today to  make use of the touch screen capabilities and have the soft keys in the menu actually be accessible by touch screen as well (in version 1.3.0, available in the next 24h or so).

The next logical step would of course be to implement touch support for the remaining functionality, e.g. selecting a menu item and scrolling up and and down a menu using swipe guestures.

So here is my question to all of you potential users of this library:  Is this something you feel would add value to this library, or are your ok with just the existing functionality?

**Comment by lvlhead** [comment]
I think it would be a useful addition that would allow devices with a touch screen and fewer than 3 buttons to be infinitely more useful (CoreS3).

**Comment by chhorisberger** [comment]
Ah good point, that makes sense. I wasn't even aware that the CoreS3 did not have those front panel buttons anymore

**Comment by chhorisberger** [comment]
Full touch support is out with version 1.4.0 (softkeys, scrolling menus, selecting menu items) - Enjoy!
